backend/src/achievements-log/achievements-log.service.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { AchievementsLog } from './entities/achievements-log.entity';
+import { AchievementsLogRepository } from './repository/achievements-log.repository';
+import { AchievementsLogDto } from './dto/achievements-log.dto';
+import { ResponseAchievementsLogDto } from './dto/response-achievements-log.dto';
+import { DeleteResult } from 'typeorm';
+
+@Injectable()
+export class AchievementsLogService {
+  constructor(
+    @InjectRepository(AchievementsLogRepository)
+    private achievementsLogRepository: AchievementsLogRepository,
+  ) {}
+
+  async findAll(): Promise<ResponseAchievementsLogDto[]> {
+    return await this.achievementsLogRepository.findAll();
+  }
+
+  async findById(id: number): Promise<AchievementsLog> {
+    return await this.achievementsLogRepository.findOneOrFail({
+      where: { id: id },
+    });
+  }
+
+  create(achievementsLogdto: AchievementsLogDto): Promise<AchievementsLog> {
+    return this.achievementsLogRepository.save(achievementsLogdto);
+  }
```

Comment:
I would consider either using CreateDateColumn (as mentioned below) or generating the `date` here in the service, rather than expecting the client to send it. This would prevent the client from falsifying the date